Description: Fix stack pointer handling in 16-bit relocator
Author: Colin Watson <cjwatson@ubuntu.com>
Bug-Ubuntu: https://bugs.launchpad.net/bugs/683904
Forwarded: yes
Applied-Upstream: http://bazaar.launchpad.net/~vcs-imports/grub/grub2-bzr/revision/3230
Last-Update: 2011-04-21

Index: b/grub-core/lib/i386/relocator16.S
===================================================================
--- a/grub-core/lib/i386/relocator16.S
+++ b/grub-core/lib/i386/relocator16.S
@@ -130,7 +130,7 @@
 	.byte	0xb8
 VARIABLE(grub_relocator16_sp)
 	.word	0
-	movw	%ax, %ss
+	movzwl	%ax, %esp
 
 	/* movw imm32, %edx.  */
 	.byte	0x66, 0xba
